pokemon-api/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/facade/BadgeFacade.java

```diff
@@ -0,0 +1,32 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.facade;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeCreateDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeStatusChangeDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.GymDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.TrainerDTO;
+
+/**
+ * Interface for Badge Facade
+ *
+ * @author Michal Mokros 456442
+ */
+public interface BadgeFacade {
+
+    Long createBadge(BadgeCreateDTO badge);
+
+    BadgeDTO findBadgeById(Long id);
+
+    GymDTO getGymOfBadge(Long id);
+
+    TrainerDTO getTrainerOfBadge(Long id);
+
+    void revokeBadgeForTrainer(Long trainerId, BadgeStatusChangeDTO badge);
+
+    void issueBadgeToTrainer(Long trainerId, BadgeStatusChangeDTO badge);
+
+    void challengeTrainer(Long trainerId, BadgeStatusChangeDTO badge);
```

Comment:
What is the use case for this - WAITING_TO_ACCEPT is the default state at creation and I don't see a reason why a badge would return to that state.

Reply:
Sorry, my bad, it is needed, however, it would be better to call it reopenChallenge or something along those lines.

Reply:
If the trainer is rechallenged for the same badge, question was asked earlier in review, we cannot have two badges with the same gym and trainer